libs/hwes/event.go

```diff
@@ -2,29 +2,124 @@ package hwes
 
 import (
 	"encoding/json"
+	"fmt"
 	"github.com/EventStore/EventStore-Client-Go/esdb"
 	"github.com/google/uuid"
+	"strings"
 	"time"
 )
 
 type Event struct {
-	ID        uuid.UUID
-	Type      string
-	Data      []byte
-	Timestamp time.Time
+	EventID       uuid.UUID
+	EventType     string
+	AggregateID   uuid.UUID
+	AggregateType AggregateType
+	Data          []byte
+	Timestamp     time.Time
+	Version       uint64
 }
 
-func NewBaseEvent(eventType string) Event {
+func NewEventBase(aggregate Aggregate, eventType string) Event {
```

Comment:
this is just a constructor for `Event`, there is no `EventBase`, nor should there be one, as an event is never wrapped, as data is set in the event object